libr/bin/format/elf/elf.c

```diff
@@ -1828,30 +1827,30 @@ ut64 Elf_(r_bin_elf_get_init_offset)(ELFOBJ *bin) {
 ut64 Elf_(r_bin_elf_get_fini_offset)(ELFOBJ *bin) {
 	ut64 entry = Elf_(r_bin_elf_get_entry_offset) (bin);
 	ut8 buf[512];
-	if (!bin) {
-		return 0LL;
+	if (!bin || entry == UT64_MAX) {
+		return UT64_MAX;
 	}
-
-	if (r_buf_read_at (bin->b, entry+11, buf, sizeof (buf)) == -1) {
+	if (r_buf_read_at (bin->b, entry + 11, buf, sizeof (buf)) == -1) {
 		bprintf ("read (get_fini)\n");
 		return 0;
 	}
 	if (*buf == 0x68) { // push // x86/32 only
-		ut64 addr;
-		memmove (buf, buf+1, 4);
-		addr = (ut64)r_read_le32 (buf);
+		memmove (buf, buf + 1, 4);
+		ut64 addr = (ut64)r_read_le32 (buf);
 		return Elf_(r_bin_elf_v2p) (bin, addr);
 	}
 	return 0;
 }
 
 ut64 Elf_(r_bin_elf_get_entry_offset)(ELFOBJ *bin) {
-	ut64 entry;
 	if (!bin) {
 		return 0LL;
 	}
-	entry = bin->ehdr.e_entry;
+	ut64 entry = bin->ehdr.e_entry;
 	if (!entry) {
+		return UT64_MAX;
+	}
```

Comment:
This close bracket changes semantics if `entry != 0`.

Reply:
Add check for ELF relocatable file (ET_REL -- this is what the test case is) or just remove the code that sets `entry` to a section start?

Reply:
that was kind of a poc, we may not have an entry0 if paddr is 0